packages/sui-widget-embedder/src-react/Widget.js

```diff
@@ -1,56 +1,62 @@
-import React, {Component} from 'react'
+/* eslint-disable no-console */
+
+import React from 'react'
 import PropTypes from 'prop-types'
 import ReactDOM from 'react-dom'
 import Context from '@s-ui/react-context'
 import {Provider as ProviderLegacy} from '@s-ui/react-domain-connector'
+import {useMount} from '@s-ui/react-hooks'
 
-export default class Widget extends Component {
-  static propTypes = {
-    children: PropTypes.element,
-    node: PropTypes.string,
-    i18n: PropTypes.object,
-    domain: PropTypes.object,
-    isVisible: PropTypes.bool,
-    browser: PropTypes.object
-  }
-
-  static defaultProps = {
-    isVisible: true
-  }
+function renderWidgetOnDOM({children, context, node}) {
+  ReactDOM.render(
+    <Context.Provider value={context}>
+      <ProviderLegacy {...context}>{children}</ProviderLegacy>
+    </Context.Provider>,
+    node
+  )
+}
 
-  componentDidMount() {
-    const {
-      node: selector,
-      children,
-      i18n,
-      domain,
-      isVisible,
-      browser
-    } = this.props
-    const node = document.querySelector(selector)
+export default function Widget({
+  browser,
+  children,
+  domain,
+  i18n,
+  isVisible = true,
+  node, // deprecated
+  selector,
+  renderMultiple = false
+}) {
+  useMount(function() {
+    const selectorToUse = selector || node
+    if (!selectorToUse) {
+      return console.warn(
+        `[Widget] You must define a selector or node (deprecated) to use the Widget`
+      )
+    }
 
-    if (!node) {
-      return console.warn(`[Widget] unable find the selector ${selector}`) // eslint-disable-line
+    const nodes = document.querySelectorAll(selectorToUse)
+    if (!nodes.length) {
+      return console.warn(
+        `[Widget] unable find nodes using selector ${selectorToUse}`
+      )
     }
+    // depending on renderMultiple, get the full array or only the first one
+    const nodesToRender = renderMultiple ? [].slice.call(nodes) : [nodes[0]]
+    // create the context object
+    const context = {browser, domain, i18n}
 
     isVisible &&
-      ReactDOM.render(
-        <Context.Provider
-          value={{
-            i18n,
-            domain,
-            browser
-          }}
-        >
-          <ProviderLegacy i18n={i18n} domain={domain} browser={browser}>
-            {children}
-          </ProviderLegacy>
-        </Context.Provider>,
-        node
-      )
-  }
+      nodesToRender.map(node => renderWidgetOnDOM({children, context, node}))
+  })
+
+  return null
+}
 
-  render() {
-    return null
-  }
+Widget.propTypes = {
+  browser: PropTypes.object,
+  children: PropTypes.element,
+  domain: PropTypes.object,
+  i18n: PropTypes.object,
+  isVisible: PropTypes.bool,
+  node: PropTypes.string
```

Comment:
As I see in the documentation, the props `children`, `isVisible` and `selector` _(aka node)_ are required

Reply:
I've improved the PropTypes and even added a deprecation to the `node` prop. :D